crates/uv/tests/tool_list.rs

```diff
@@ -111,6 +119,9 @@ fn tool_list_missing_receipt() {
 
     ----- stderr -----
     warning: Ignoring malformed tool `black` (run `uv tool uninstall black` to remove)
+    No tools installed.
+
+    See `uv tool install --help` for more information.
```

Comment:
I think these could be displayed on the same line. Should we be changing this in this pull request though? It seems like a separate change.

Similarly, we'll need to discuss if we should say "No tools installed" when there are malformed tools installed.

Reply:
You could probably move all the changes to `uv tool list` behavior out of this and into a separate pull request.

Reply:
> Similarly, we'll need to discuss if we should say "No tools installed" when there are malformed tools installed.

Shall we display a different message? Basically, I want to avoid the case where all the tools are malformed and say here are the installed tools, but they are empty.

Reply:
Ah I see. Now I understand why you're tweaking `uv tool list`. We probably don't want to see information about malformed tools in the `uv tool run` help at all. This sort of suggests we need to stop calling the `uv tool list` implementation directly? I think generally that will make things easier, i.e. if there are no installed tools the `uv tool run` implementation can handle that directly instead.

Reply:
(this probably requires some refactor to `uv tool list` so we can avoid duplicating most of what it does?)

Reply:
Yes, this makes sense now. Let me try to refactor the code a little.